pandas/tests/groupby/test_bin_groupby.py

```diff
@@ -25,6 +25,16 @@ def test_series_grouper():
     tm.assert_almost_equal(counts, exp_counts)
 
 
+def test_series_grouper_requires_nonempty_raises():
+    # GH#29500
+    obj = Series(np.random.randn(10))
+    dummy = obj[:0]
+    labels = np.array([-1, -1, -1, 0, 0, 0, 1, 1, 1, 1], dtype=np.int64)
+
+    with pytest.raises(ValueError, match="SeriesGrouper requires non-empty `series`"):
```

Comment:
Is this visible from an end user perspective?

Reply:
this should never be raised to the user, just seemed like the behavior change merited a test.